src/main/java/com/iexec/core/worker/WorkerController.java

```diff
@@ -85,6 +85,7 @@ public ResponseEntity<String> getToken(@RequestParam(name = "walletAddress") Str
 
         String challenge = challengeService.getChallenge(walletAddress);
         byte[] hashToCheck = Hash.sha3(BytesUtils.stringToBytes(challenge));
+        challengeService.removeChallenge(walletAddress, challenge);
```

Comment:
Challenge must be removed after its verification, otherwise Mallory could attack Alice challenge by impersonating her while spamming /login

Reply:
Fixed in 8349102